### PR TITLE
Fixed type checking errors in Pytorch - torch/distributed/rpc/options.py

### PR DESCRIPTION
**Pull request by Onyiee**
Fixes #{64}
This PR fixes the type checking errors in torch/distributed/rpc/options.py. 
The variable types in 84:8 and 85:8 were  declared to have type `List`  but were sometimes assigned a value of  `None`. This caused an incompatitble variable type error. Therefore, I changed the type from `List` to `Optional[List]` . Hence, this fixes the incompatitble variable type error. 

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### CODE REVIEW

File: torch/distributed/rpc/options.py [hunk 1]
```diff
@@ -81,17 +83,15 @@ def __init__(
         init_method: str = rpc_contants.DEFAULT_INIT_METHOD,
         device_maps: Optional[Dict[str, Dict[DeviceType, DeviceType]]] = None,
         devices: Optional[List[DeviceType]] = None,
-        _transports: List = None,
-        _channels: List = None,
+        _transports: Optional[List] = None,
+        _channels: Optional[List] = None,
```
Comment (mrshenli) [inline torch/distributed/rpc/options.py:87]: Would I be correct if these two lines are the only non-formatting change in this file?

**Review by mrshenli — COMMENTED** [review]
Thanks! Changes in `options.py` LGTM!

Could you please remove the changes in `fx/fuse.py` from this PR and create a new PR for it? That change does not match the title and will require a different team to review, thanks!

**Review by mrshenli — COMMENTED** [review]
The PR summary mentioned "Fixes #{64}". I am assume this is not intentional?

**Review by mrshenli — COMMENTED** [review]
BTW, there are conflicts with master. Could you please rebase?

File: torch/distributed/rpc/options.py (hunk 1 above)
Comment (onionymous) [reply torch/distributed/rpc/options.py:87]: Yep! For context, this is part of a project to migrate Pytorch from Mypy to Pyre. Pytorch requires `Optional` types to be explicitly specified while Mypy will automatically infer it as `Optional` if it takes a value of `None`.

**Comment by Onyiee** [comment]
I am closing this PR because of the complications around it since I didn’t create a new branch before making my changes. 

Here's a link to the new PR - https://github.com/pytorch/pytorch/compare/master...Onyiee:Onyiee2?expand=1

**Pull request closed without merging**